### DISCUSSION TITLE
How to disable changing a column in migration?

### DISCUSSION THREAD

**Discussion by itxtoledo** [mikro-orm/mikro-orm#4647] [Q&A]
I'm leaving sequelize and migrating to mikroorm.

I have the following entity:

```TypeScript
import { Entity, PrimaryKey, Property } from "@mikro-orm/core";

@Entity({ tableName: "Clicks" })
export class Click {
  @PrimaryKey({
    type: "int",
    columnType: "int",
  })
  id!: number;

  @Property({ type: "varchar(255)", columnType: "varchar(255)" })
  target!: string;

  @Property({ type: "varchar(255)", columnType: "varchar(255)" })
  medium!: string;

  @Property({ type: "varchar(255)", columnType: "varchar(255)" })
  campaign!: string;

  @Property({
    name: "createdAt",
    type: "datetime",
    columnType: "datetime",
  })
  createdAt?: Date = new Date();

  @Property({
    name: "updatedAt",
    type: "datetime",
    columnType: "datetime",
    onUpdate: () => new Date(),
  })
  updatedAt?: Date = new Date();
}
``` 

and my database schema is like this (generated with sequelize)

![image](https://github.com/mikro-orm/mikro-orm/assets/21026675/d86e5363-4221-4037-a2c5-68f8702ff15a)

```SQL
CREATE TABLE `Clicks` (
  `id` int(11) NOT NULL,
  `target` varchar(255) NOT NULL,
  `medium` varchar(255) NOT NULL,
  `campaign` varchar(255) NOT NULL,
  `createdAt` datetime NOT NULL,
  `updatedAt` datetime NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```

But every time I run the command to create a migration, mikroorm writes this:

```TypeScript
this.addSql('alter table `Clicks` modify `id` int unsigned not null;');
```

I tried to add this from the documentation but mikroorm keeps trying to change the entity id column

```TypeScript
ignoreSchemaChanges: ['type', 'extra'],
```

**Comment by B4nan** [comment]
So your PK column is signed integer? Then just add `unsigned: false` to the options, as the default is to map to unsigned int. 

You can't disable this, the `ignoreSchemaChanges` has a pretty limited use (and the use case is around unsupported mapping like virtual columns, not wrong mapping like this). What you should do instead is to define the mapping properly.

Note that you also don't need the `columnType`, nor the `type` (if you use reflect-metadata, the default). All the options you specified are the default mapping (`number`s are mapped to `integer`s, `string`s are mapped to `varchar(255)`). It makes no sense to just repeat the same thing in both.

Try this:

```ts
@PrimaryKey({ unsigned: false })
id!: number;
```

Or just update your schema, using signed int for a PK seems pretty weird in the first place, your schema is not set in stone, right? :]

FYI there is one hidden feature you can use to see why the schema diffing sees something as changed, just enable `MIKRO_ORM_VERBOSE` env var when using the CLI:

```bash
MIKRO_ORM_VERBOSE=1 npx mikro-orm schema:update -d
```
